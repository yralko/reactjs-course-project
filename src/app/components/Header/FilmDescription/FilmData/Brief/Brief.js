import React from 'react';
import classes from './index.css';

const Brief = props => (
  <div className={classes.Brief}>
    <div className={classes.upperRow}>
      <h2>Film name {props.title}</h2>
      <span className={classes.filmRating}>{props.rating}</span>
    </div>
    <p className={classes.genres}>
      {props.genres.length > 1
        ? `${props.genres.slice(0, -1)} & ${props.genres.slice(-1)}`
        : props.genres}
    </p>
    <div className={classes.lowerRow}>
      <span>{props.year.slice(0, 4)}</span>
      <span>{props.duration} min</span>
    </div>
  </div>
);

export default Brief;
