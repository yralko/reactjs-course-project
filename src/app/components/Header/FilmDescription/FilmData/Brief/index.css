.Brief .upperRow {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-top: 10px;
}

.Brief .upperRow > * {
  margin: 0;
}

.Brief .upperRow h2 {
  color: #F65260;
  margin-right: 20px;
}

.filmRating {
  padding: 7px;
  border: 1px solid #ffffff;
  border-radius: 50%;
  color: #ffffff;
}

.genres {
  color: #eeeeee;
  margin-top: 5px;
  margin-bottom: 15px;
}

.lowerRow {
  color: #ffffff;
  font-weight: bold;
}

.lowerRow span:first-child {
  margin-right: 20px;
}
