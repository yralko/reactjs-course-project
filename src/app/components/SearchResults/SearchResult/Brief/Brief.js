import React from 'react';
import classes from './index.css';

const Brief = props => (
  <div className={classes.Brief}>
    <div className={classes.upperRow}>
      <h4 className={classes.title}>{props.filmTitle}</h4>
      <span className={classes.date}>{props.date}</span>
    </div>
    <p className={classes.genres}>{props.genres}</p>
  </div>
);

export default Brief;
