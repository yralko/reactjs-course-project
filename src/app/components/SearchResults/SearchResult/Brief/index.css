.Brief {
  padding: 7px;
}

.upperRow {
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
}

.upperRow > * {
  margin: 5px 0;
}

.upperRow .title {
  text-transform: uppercase;
}

.upperRow .date {
  padding: 2px 5px;
  font-size: 11px;
  border: 1px solid #c5c5c5;
  border-radius: 10%;
}

.genres {
  font-size: 12px;
  margin-top: 5px;
}
