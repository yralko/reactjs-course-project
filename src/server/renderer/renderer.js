import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter as Router } from 'react-router-dom';
import App from '../../app/App';


export default (req, store, context) => {
  const root = (
    <Provider store={store}>
      <Router location={req.url} context={context}>
        <App />
      </Router>
    </Provider>
  );

  const content = renderToString(root); 

  return `
  <!doctype html>
  <html lang="en">
  <head>
    <meta charset="utf-8">
    <title>The HTML5 Herald</title>
    <base href="/" />
    <link rel="stylesheet" href="style.css" />
  </head>
  <body>
    <div id="root">${content}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState())}</script>
    <script src="bundle.js"></script>
  </body>
  </html>
  `;
};
