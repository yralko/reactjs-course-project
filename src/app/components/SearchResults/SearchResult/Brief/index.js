import Brief from './Brief';

export default Brief;
