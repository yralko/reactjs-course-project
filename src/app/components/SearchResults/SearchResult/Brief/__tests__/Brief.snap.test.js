import React from 'react';
import renderer from 'react-test-renderer';
import Brief from '../Brief';

test('matches snapshot', () => {
  const tree = renderer.create(<Brief />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
