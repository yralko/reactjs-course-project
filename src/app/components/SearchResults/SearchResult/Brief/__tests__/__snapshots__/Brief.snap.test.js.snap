// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <h4
      className={undefined}
    />
    <span
      className={undefined}
    />
  </div>
  <p
    className={undefined}
  />
</div>
`;
